### PR TITLE
Tests for table component

### PR DESCRIPTION
**Pull request by wenche**
Part of #297

### CODE REVIEW

**Comment by wenche** [comment]
Don't test implementation details they say 🙈

File: libraries/core-react/src/components/Table/Table.test.tsx [hunk 1]
```diff
@@ -41,3 +55,72 @@ describe('Caption', () => {
...
+          <Row>
+            <Cell variant="numeric">{text}</Cell>
+          </Row>
+        </Body>
+      </Table>,
+    )
+
+    expect(getByText(text)).toHaveStyleRule(
+      'font-feature-settings',
+      "'tnum' on,'lnum' on",
```
Comment (mimarz) [inline libraries/core-react/src/components/Table/Table.test.tsx:114]: This should be tested against the `DataCell` tokens value for numeric cells

File: libraries/core-react/src/components/Table/Table.test.tsx [hunk 2]
```diff
@@ -41,3 +55,72 @@ describe('Caption', () => {
...
+          <Row>
+            <Cell>{body}</Cell>
+          </Row>
+        </Body>
+      </Table>,
+    )
+    const headElement = getByText(header).closest('thead')
+    const regularContent = getByText(body)
+    expect(headElement).not.toHaveStyleRule('background', 'rgba(255,255,255,1)')
+    expect(regularContent).toHaveStyleRule('background', 'rgba(255,255,255,1)')
```
Comment (mimarz) [inline libraries/core-react/src/components/Table/Table.test.tsx:97]: If I am not mistaken the Table head also has a thicker bottom border that can be asserted on.

**Review by mimarz — CHANGES_REQUESTED** [review]
We still have some missing features in `Table` that needs to have testes further on but its a start :)

File: libraries/core-react/src/components/Table/Table.test.tsx (hunk 1 above)
Comment (wenche) [reply libraries/core-react/src/components/Table/Table.test.tsx:114]: Yes, probably... But it gives 
Skjermbilde 2020-12-14 kl  10 27 59

Comment (wenche) [reply libraries/core-react/src/components/Table/Table.test.tsx]: Tried to fix this in figma-broker, but it didn't work that well... Seems like prettier is re adding the whitespace or something 🤯  Cheated and  trimmed the text. I'm not sure if the world is a better place or not after this change 🤷‍♀️  @mimarz

File: libraries/core-react/src/components/Table/Table.test.tsx (hunk 2 above)
Comment (wenche) [reply libraries/core-react/src/components/Table/Table.test.tsx]: I added that as well. Using the tokens as assertion was just too complicated so I just added the border as a string. But things are slightly better by the usage of tokens for the white colour at least

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**